app/scripts/controllers/exportImport.js

```diff
@@ -198,122 +213,36 @@ angular.module('openhimConsoleApp')
-      $scope.importProgressStatus = 0;
-      $scope.importProgressType = '';
-
-      // convert to json object
-      data = JSON.parse(data);
-
-      var totalRecords = 0;
-      if (data.Clients) {
-        totalRecords += data.Clients.length;
-      }
-      if (data.Channels) {
-        totalRecords += data.Channels.length;
-      }
-      if (data.ContactGroups) {
-        totalRecords += data.ContactGroups.length;
-      }
-      if (data.Users) {
-        totalRecords += data.Users.length;
-      }
-      if (data.Mediators) {
-        totalRecords += data.Mediators.length;
-      }
-
-      var doneItems = 0;
-
-      //loop through each collection in object
-      angular.forEach(data, function(modelRecords, model) {
-
-        // loop through each record
-        angular.forEach(modelRecords, function(value) {
-
-          var record;
-          var insertParams = {};
-
-          // check model to determine which API to call
-          switch(model) {
-            case 'Clients':
-              record = new Api.Clients( value );
-              break;
-            case 'Channels':
-              record = new Api.Channels( value );
-              break;
-            case 'ContactGroups':
-              record = new Api.ContactGroups( value );
-              break;
-            case 'Mediators':
-              if ( value._id ){
-                delete value._id;
-              }
-              record = new Api.Mediators( value );
-              insertParams.urn = '';
-              break;
-            case 'Users':
-              record = new Api.Users( value );
-              insertParams.email = '';
-              break;
-            default:
-              // no default yet
-          }
-
-          // if record has _id then do update
-          if ( record._id ){
-            record.$update(function(){
-              importSuccess();
-            }, function(err){
-              importFail(model, value, err);
-            });
-          }else{
-            record.$save(insertParams, function(){
-              importSuccess();
-            }, function(err){
-              importFail(model, value, err);
-            });
-          }
+    var validateSuccess = function(result) {
+      console.log('succesfully validated uploaded file');
+      $scope.importStatus = 'resolvingConflicts';
+      $scope.validatedData = result;
 
-          doneItems++;
-          $scope.importProgressStatus = Math.floor( doneItems / totalRecords );
-
-          // update progress bar too 100%
-          if( doneItems === totalRecords ){
-            $scope.importProgressStatus = 100;
-            $scope.importProgressType = 'success';
-          }
+      openValidationModal();
+    };
 
-        });
+    $scope.validateImportFile = function(data) {
+      $scope.importStatus = 'progress';
 
+      Api.MetadataValidation.save(data, function(result){
+        validateSuccess(result);
+      }, function(err){
+        validateFail(err);
```

Comment:
This can be simplified to `Api.MetadataValidation.save(data, validateSuccess, vallidateFail)`